### PR TITLE
screen: refactor old curwin-heavy logic and graduate "msgsep" feature

### PR DESCRIPTION
**Pull request by bfredl**

### CODE REVIEW

File: test/functional/legacy/cmdline_spec.lua [hunk 1]
```diff
@@ -185,7 +185,6 @@ describe('cmdline', function()
       set laststatus=2
       set statusline=%=:%{getcmdline()}
       autocmd CmdlineChanged * redrawstatus
-      set display-=msgsep
     ]])
     -- :redrawstatus is postponed if messages have scrolled
     feed([[:echo "one\ntwo\nthree\nfour"<CR>]])
```
Comment (zeertzjq) [inline test/functional/legacy/cmdline_spec.lua:190]: This whole test can be removed.

File: runtime/doc/vim_diff.txt [hunk 2]
```diff
@@ -38,7 +38,6 @@ centralized reference of the differences.
 - 'complete' excludes "i"
 - 'cscopeverbose' is enabled
 - 'directory' defaults to ~/.local/state/nvim/swap// (|xdg|), auto-created
-- 'display' defaults to "lastline,msgsep"
```
Comment (justinmk) [inline runtime/doc/vim_diff.txt:41]: "lastline" still differs from vim, no?

Also should have a "UI" section somewhere in `nvim-features` which mentions what used to be described in the `msgsep` help doc.

File: src/nvim/move.c [hunk 3]
```diff
@@ -132,13 +132,6 @@ static void redraw_for_cursorcolumn(win_T *wp)
 
 // Update curwin->w_topline and redraw if necessary.
 // Used to update the screen before printing a message.
```
Comment (zeertzjq) [inline src/nvim/move.c:134]: Also remove this comment?

Comment (zeertzjq) [inline src/nvim/move.c:134]:
```suggestion
```

File: runtime/doc/ui.txt [hunk 4]
```diff
@@ -611,8 +611,8 @@ tabs.
 	Display messages on `grid`.  The grid will be displayed at `row` on the
 	default grid (grid=1), covering the full column width. `scrolled`
 	indicates whether the message area has been scrolled to cover other
-	grids. It can be useful to draw a separator then ('display' msgsep
-	flag). The Builtin TUI draws a full line filled with `sep_char` and
+	grids. It can be useful to draw a separator then ('fillchars' msgsep
+	field). The Builtin TUI draws a full line filled with `sep_char` and
```
Comment (zeertzjq) [inline runtime/doc/ui.txt:615]: Maybe just using `|msgsep|` here is better?

**Pull request merged**